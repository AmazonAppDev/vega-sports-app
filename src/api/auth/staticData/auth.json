{
  "auth": {
    "id": "nmbsd-234sdc-678-jklsdf",
    "name": "Freida",
    "surname": "Gomam",
    "email": "fgomam@fakemail.com",
    "address": "Nowhere 5 USA",
    "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/5p.jpg",
    "profiles": [
      {
        "id": "bada55-1dl00k-4tm33",
        "displayName": "Freida",
        "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/6p.jpg"
      },
      {
        "id": "l00k-1nt0-s0uL",
        "displayName": "Jennifer",
        "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/7p.jpg"
      }
    ]
  }
}
