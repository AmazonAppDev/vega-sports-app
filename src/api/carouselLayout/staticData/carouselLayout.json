{
  "carousellayout": [
    {
      "id": "e77effd2-60fd-4a75-9921-ee1596c4d248",
      "carouselType": "hero",
      "carouselTitle": null,
      "endpoint": "suggestedforyou"
    },
    {
      "id": "fac0db1d-60ee-4ff9-9c3f-fc1ca24fa6be",
      "carouselType": "square",
      "carouselTitle": "Live TV: fresh news from events",
      "endpoint": "livestreams"
    },
    {
      "id": "cf5bafe0-023b-4ff9-9c3f-fc1ca24fa6be",
      "carouselType": "card",
      "carouselTitle": "Sport Teams Hall of Fame",
      "endpoint": "teams"
    },
    {
      "id": "068f3841-023b-4de8-85d1-f410d1407528",
      "carouselType": "square",
      "carouselTitle": "Documentaries on Sport Stars",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc13a87-af56-42ef-a66e-b882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f1389-845c-476d-820b-1aa388e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d21039f-3ba5-491b-b481-324ceb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "fac0db1d-60ee-4ff9-9c3f-fc15ca24fa6be",
      "carouselType": "square",
      "carouselTitle": "Live TV: fresh news from events",
      "endpoint": "livestreams"
    },
    {
      "id": "cf5bafe0-023b-4ff9-9c3f-6fc1ca24fa6be",
      "carouselType": "card",
      "carouselTitle": "Sport Teams Hall of Fame",
      "endpoint": "teams"
    },
    {
      "id": "06844f3841-023b-4de8-85d1-7f410d14aa07528",
      "carouselType": "square",
      "carouselTitle": "Documentaries on Sport Stars",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc1113a87-af56-42ef-a66e8-b8jj8a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f11389-845c-476d-8290b-1aajj88e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d221039f-3ba5-491b-b4081-32cejjb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc132a87-af56-42ef-a66e-bgh882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c9177f14389-845c-476d-820b-s",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "9fc166113a87-af56-42ef-a66e8-b8jj8a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f5511389-845c-476d-8290b-1aajj88e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d22441039f-3ba5-491b-b4081-32cejjb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc13233a87-af56-42ef-a66e-bgh882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f1422389-845c-476d-820b-s",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    }
  ]
}
