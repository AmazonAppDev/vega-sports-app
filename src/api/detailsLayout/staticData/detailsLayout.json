{
  "detailslayout": [
    {
      "id": "documentaries",
      "layout_type": "documentaries",
      "layout_title": "Sports Documentaries",
      "layoutElements": [
        {
          "layoutElements": [
            {
              "layoutElements": [
                {
                  "id": "containera-imagetile1",
                  "description": "Explore the journey of legendary athletes.",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/30.jpg"
                  },
                  "displayProps": {
                    "size": "custom",
                    "customWidth": 500,
                    "customHeight": 200
                  },
                  "targetUrl": {
                    "url": "details/documentaries"
                  },
                  "title": "Athlete Stories",
                  "elementType": "DCImageTile"
                },
                {
                  "id": "containera-imagetile2",
                  "description": "Read about the top stars.",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/31.jpg"
                  },
                  "displayProps": {
                    "size": "custom",
                    "customWidth": 500,
                    "customHeight": 200
                  },
                  "targetUrl": {
                    "url": "details/documentaries"
                  },
                  "title": "Top stars",
                  "elementType": "DCImageTile"
                }
              ],
              "displayProps": {
                "horizontalSpacing": 0,
                "flexDirection": "row"
              },
              "title": null,
              "elementType": "DCContentContainer"
            },
            {
              "layoutElements": [
                {
                  "description": "A closer look at iconic sports moments.",
                  "displayProps": {
                    "size": "small"
                  },
                  "id": "containera-subcontainer-imagetile1",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/32.jpg"
                  },
                  "targetUrl": {
                    "url": "details/documentaries"
                  },
                  "title": "Iconic Moments",
                  "elementType": "DCImageTile"
                },
                {
                  "description": "Behind the scenes of championship games.",
                  "displayProps": {
                    "size": "small"
                  },
                  "id": "containera-subcontainer-imagetile2",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/33.jpg"
                  },
                  "targetUrl": {
                    "url": "details/documentaries"
                  },
                  "title": "Championship Stories",
                  "elementType": "DCImageTile"
                },
                {
                  "description": "The untold stories of rising stars.",
                  "displayProps": {
                    "size": "small"
                  },
                  "id": "containera-subcontainer-imagetile3",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/35.jpg"
                  },
                  "targetUrl": {
                    "url": "details/documentaries"
                  },
                  "title": "Rising Stars",
                  "elementType": "DCImageTile"
                },
                {
                  "description": "Team insights and player stats.",
                  "displayProps": {
                    "size": "small"
                  },
                  "id": "sectionb-tilewithimage2",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/36.jpg"
                  },
                  "targetUrl": {
                    "url": "details/teams"
                  },
                  "title": "Team Profiles",
                  "elementType": "DCImageTile"
                }
              ],
              "displayProps": {
                "horizontalSpacing": 18,
                "flexDirection": "row",
                "verticalSpacing": 20
              },
              "title": null,
              "elementType": "DCContentContainer"
            },
            {
              "elementType": "DCText",
              "text": "Athlete stories provide an insight into the lives of sports stars beyond the game. They highlight the challenges, sacrifices, and triumphs that shape their journeys to success. From overcoming injuries to breaking records, these stories inspire fans and aspiring athletes alike. They often reveal the personal struggles and resilience that drive their passion for the sport. Athlete stories also celebrate their achievements, both on and off the field, showcasing their impact on the community. Through these narratives, we connect with the human side of sports, finding motivation and admiration in their experiences.",
              "displayProps": {
                "variant": "body",
                "alignContent": "center",
                "alignItems": "center",
                "justifyContent": "center"
              },
              "id": "container-body1"
            }
          ],
          "displayProps": {
            "horizontalSpacing": 18
          },
          "title": "Documentary Highlights",
          "elementType": "DCContentContainer"
        }
      ]
    },
    {
      "id": "livestreams",
      "layout_type": "livestreams",
      "layout_title": "Live Matches",
      "layoutElements": [
        {
          "layoutElements": [
            {
              "description": "Team 1 is ready to dominate the field.",
              "displayProps": {
                "size": "medium"
              },
              "id": "containerlivestreams-imagetile1",
              "image": {
                "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/38.jpg",
                "targetUrl": "team1.thumbnail"
              },
              "targetUrl": {
                "url": "details/teams"
              },
              "title": "Team 1",
              "titleTarget": "team1.name",
              "elementType": "DCImageTile"
            },
            {
              "displayProps": {
                "variant": "headline",
                "alignContent": "center",
                "alignItems": "center",
                "justifyContent": "center"
              },
              "id": "containerlivestreams-text1",
              "text": "VS",
              "elementType": "DCText"
            },
            {
              "description": "Team 2 is bringing their A-game.",
              "id": "containerlivestreams-imagetile2",
              "displayProps": {
                "size": "medium"
              },
              "image": {
                "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/39.jpg",
                "targetUrl": "team2.thumbnail"
              },
              "targetUrl": {
                "url": "details/teams"
              },
              "title": "Team 2",
              "titleTarget": "team2.name",
              "elementType": "DCImageTile"
            }
          ],
          "displayProps": {
            "horizontalSpacing": 18,
            "flexDirection": "row"
          },
          "title": null,
          "elementType": "DCContentContainer"
        },
        {
          "layoutElements": [
            {
              "displayProps": {
                "variant": "headline"
              },
              "id": "containerlivestreams-text1",
              "text": "Upcoming Match Highlights",
              "textTarget": "headline",
              "elementType": "DCText"
            },
            {
              "displayProps": {
                "variant": "body"
              },
              "id": "containerlivestreams-text2",
              "text": "Get ready for an exciting clash between two top teams. Stay tuned for live updates and in-depth analysis.",
              "textTarget": "description",
              "elementType": "DCText"
            }
          ],
          "displayProps": {
            "horizontalSpacing": 50,
            "verticalSpacing": 30
          },
          "title": null,
          "elementType": "DCContentContainer"
        }
      ]
    },
    {
      "id": "suggestedforyou",
      "layout_type": "suggestedforyou",
      "layout_title": "Recommended for You",
      "layoutElements": [
        {
          "layoutElements": [
            {
              "displayProps": {
                "variant": "headline"
              },
              "id": "containerlivestreams-text1",
              "text": "Curated Sports Content Just for You",
              "elementType": "DCText"
            }
          ],
          "title": null,
          "elementType": "DCContentContainer"
        }
      ]
    },
    {
      "id": "teams",
      "layout_type": "teams",
      "layout_title": "Team Profiles",
      "layoutElements": [
        {
          "id": "containerteams2",
          "layoutElements": [
            {
              "id": "containerteams2-1",
              "layoutElements": [
                {
                  "description": "",
                  "displayProps": {
                    "size": "custom",
                    "customWidth": 300,
                    "customHeight": 450
                  },
                  "id": "containerteams-imagetile1",
                  "image": {
                    "targetUrl": "thumbnail"
                  },
                  "titleTarget": "teamName",
                  "targetUrl": {
                    "url": "details/teams"
                  },
                  "elementType": "DCImageTile"
                }
              ],
              "displayProps": {
                "horizontalSpacing": 18,
                "flex": 0
              },
              "title": null,
              "elementType": "DCContentContainer"
            },
            {
              "id": "containerteams2-2",
              "layoutElements": [
                {
                  "displayProps": {
                    "variant": "body"
                  },
                  "id": "containerteams2-text1",
                  "text": "Sport type:",
                  "title": "DCText 1",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "title",
                    "horizontalSpacing": 24
                  },
                  "id": "containerteams2-text2",
                  "text": "N/A",
                  "textTarget": "sportType",
                  "title": "DCText 2",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "body",
                    "topSpacing": 24
                  },
                  "id": "containerteams2-text3",
                  "text": "Coach:",
                  "title": "DCText 3",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "title",
                    "horizontalSpacing": 24
                  },
                  "id": "containerteams2-text4",
                  "text": "N/A",
                  "textTarget": "coachName",
                  "title": "DCText 4",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "body",
                    "topSpacing": 24
                  },
                  "id": "containerteams2-text5",
                  "text": "Home stadium:",
                  "title": "DCText 5",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "title",
                    "horizontalSpacing": 24
                  },
                  "id": "containerteams2-text6",
                  "text": "N/A",
                  "textTarget": "homeStadium",
                  "title": "DCText 6",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "body",
                    "topSpacing": 24
                  },
                  "id": "containerteams2-text7",
                  "text": "Team color:",
                  "title": "DCText 7",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "title",
                    "horizontalSpacing": 24
                  },
                  "id": "containerteams2-text8",
                  "text": "N/A",
                  "textTarget": "teamColor",
                  "title": "DCText 8",
                  "elementType": "DCText"
                }
              ],
              "displayProps": {
                "alignSelf": "flex-start",
                "justifyContent": "flex-start",
                "alignContent": "flex-start"
              },
              "title": null,
              "elementType": "DCContentContainer"
            }
          ],
          "displayProps": {
            "horizontalSpacing": 18,
            "verticalSpacing": 36,
            "flexDirection": "row",
            "alignSelf": "flex-start",
            "justifyContent": "flex-start",
            "alignContent": "flex-start"
          },
          "title": null,
          "elementType": "DCContentContainer"
        }
      ]
    }
  ]
}
