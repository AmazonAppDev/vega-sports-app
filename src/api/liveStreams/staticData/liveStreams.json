{
  "livestreams": [
    {
      "id": "50128bae-e954-4233-8e15-cd5867a31370",
      "stream_date": "2022-12-17T04:52:43Z",
      "title": "Live from Century's Football Match",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/11.jpg",
      "sport_type": "surfing"
    },
    {
      "id": "d9f25760-7263-426a-b4d7-a25e14a26f03",
      "stream_date": "2022-06-09T11:58:23Z",
      "title": "Formula 1 Discussion Panel",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/12.jpg",
      "sport_type": "soccer"
    },
    {
      "id": "73780c3d-2f09-4ea0-b9c0-50bdf2a9ba0a",
      "stream_date": "2022-10-07T23:10:08Z",
      "title": "Inside the Championship Studio",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/13.jpg",
      "sport_type": "padel"
    },
    {
      "id": "4448fb6d-f55c-4975-8b0c-14d5356d807b",
      "stream_date": "2022-01-25T06:12:26Z",
      "title": "Post-Match Analysis Desk",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/14.jpg",
      "sport_type": "surfing"
    },
    {
      "id": "997afe05-d0fa-44c2-b62e-555062807623",
      "stream_date": "2022-08-20T07:38:40Z",
      "title": "Voices from the Sidelines",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/15.jpg",
      "sport_type": "surfing"
    },
    {
      "id": "dad18f0f-d753-47b7-9393-e30630a394cd",
      "stream_date": "2022-08-27T15:58:41Z",
      "title": "Breaking News: Sports Edition",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/16.jpg",  
      "sport_type": "skiing"
    },
    {
      "id": "18813033-6113-4da8-ad75-2a209da6380f",
      "stream_date": "2024-10-17T20:05:31Z",
      "title": "Live Debate: Game Changers",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/17.jpg",
      "sport_type": "football"
    },
    {
      "id": "449e40c9-1b3a-4a33-931f-5172b5abaf39",
      "stream_date": "2022-08-11T07:26:11Z",
      "title": "Studio Insights: Race Day",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/18.jpg",
      "sport_type": "marathon"
    },
    {
      "id": "386e49d7-3ff1-476a-82b9-b9a61870b3b3",
      "stream_date": "2022-04-30T05:45:24Z",
      "title": "Reporter’s Corner: Match Highlights",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/19.jpg",
      "sport_type": "running"
    },
    {
      "id": "9fbaaba2-b6d0-442e-83fe-529212062a4c",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "On-Air with the Experts",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/20.jpg",
      "sport_type": "soccer"
    },
    {
      "id": "c4253727-d8fe-4b22-b792-e1c79ecaa5f4",
      "title": "In-Studio with Top Analysts",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/24.jpg",
      "sport_type": "boxing"
    },
    {
      "id": "f4a39998-4937-4bb7-82e7-90108a12a702",
      "stream_date": "",
      "title": "Game Day Reporter Spotlight",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/25.jpg",
      "sport_type": "football"
    },
    {
      "id": "39abb98c-922d-4f10-b205-4b95a4d4babb",
      "stream_date": null,
      "title": "Behind the Mic: Sports Talk",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/26.jpg",
      "sport_type": "biking"
    },
    {
      "id": "967d3e67-473d-4351-9c10-3727f55297ee",
      "stream_date": "2022-08-27T15:58:41Z",
      "title": "Breaking News: Sports Edition",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/27.jpg",
      "sport_type": "running"
    },
    {
      "id": "074ac572-c778-4402-bd68-61438f28e10f",
      "stream_date": "2024-10-17T20:05:31Z",
      "title": "Live Debate: Game Changers",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/28.jpg",
      "sport_type": "basketball"
    },
    {
      "id": "c9027268-5ef4-4663-b979-a8ae527373a1",
      "stream_date": "2022-08-11T07:26:11Z",
      "title": "Studio Insights: Race Day",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/29.jpg",
      "sport_type": "baseball"
    },
    {
      "id": "7e1a09a4-1066-43d3-9ee9-0eb89ade5c18",
      "stream_date": "2022-04-30T05:45:24Z",
      "title": "Reporter’s Corner: Match Highlights",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/30.jpg",
      "sport_type": "land hockey"
    },
    {
      "id": "1eeadc2b-4e6f-461a-8f93-825ef54b2f48",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "On-Air with the Experts",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/31.jpg",
      "sport_type": "tennis"
    },
    {
      "id": "144509c0-424f-40e3-ad26-f1ae885e8b79",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "Pre-Game Strategy Breakdown",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/32.jpg",
      "sport_type": "running"
    },
    {
      "id": "93e85028-3a15-44e1-b976-813711a6b938",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "Final Whistle Recap Show",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/33.jpg",
      "sport_type": "biking"
    }
  ]
}