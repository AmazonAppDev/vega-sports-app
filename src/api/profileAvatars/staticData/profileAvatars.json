{
  "profileavatars": [
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/1p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/2p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/3p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/4p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/5p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/6p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/7p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/8p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/9p.jpg"
  ]
}
