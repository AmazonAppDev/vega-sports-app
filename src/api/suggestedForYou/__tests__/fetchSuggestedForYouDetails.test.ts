import { useQuery } from '@tanstack/react-query';
import { renderHook } from '@testing-library/react-native';

import { AppConfig } from '@AppServices/appConfig';
import {
  fetchSuggestedForYouDetailsApiCall,
  useSuggestedForYouDetails,
} from '../fetchSuggestedForYouDetails';
import staticData from '../staticData/suggestedForYou.json';

jest.mock('@tanstack/react-query');

const mockRQ = useQuery as jest.Mock;

describe('fetchSuggestedForYouDetailsApiCall', () => {
  beforeAll(() => {
    require('jest-fetch-mock').enableMocks();
  });

  afterEach(() => {
    fetchMock.resetMocks();
  });

  describe('fetchSuggestedForYouDetailsApiCall', () => {
    it('should return parsed suggested for you data on success', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.suggestedforyou }),
        {
          status: 200,
        },
      );

      const result = await fetchSuggestedForYouDetailsApiCall(
        'e9a06a8f-9d40-41b9-a8b4-38bbc67159a2',
      );

      expect(result).toMatchSnapshot();
    });

    it('should return parsed airDate', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.suggestedforyou }),
        {
          status: 200,
        },
      );

      const result = await fetchSuggestedForYouDetailsApiCall('empty-air-date');

      expect(result).toMatchSnapshot();
    });

    it('should return undefined for unexisting id', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.suggestedforyou }),
        {
          status: 200,
        },
      );

      const result = await fetchSuggestedForYouDetailsApiCall('unexisting-id');

      expect(result).toBeUndefined();
    });

    test.each([{ itemId: '' }, { itemId: null }, { itemId: undefined }])(
      'should return undefined if suggested for you details data has nullable id ($itemId)',
      async ({ itemId }) => {
        await expect(
          // @ts-expect-error intentionally break TS
          fetchSuggestedForYouDetailsApiCall(itemId),
        ).rejects.toThrow(
          'fetchSuggestedForYouDetailsApiCall() was used with invalid item id',
        );
      },
    );

    it('should throw error for 400 status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      fetchMock.mockResponseOnce(JSON.stringify({ suggestedforyou: {} }), {
        status: 400,
      });

      await expect(
        fetchSuggestedForYouDetailsApiCall(
          'e9a06a8f-9d40-41b9-a8b4-38bbc67159a2',
        ),
      ).rejects.toThrow(
        `fetchSuggestedForYouDetailsApiCall(): resources does not exists for endpoint 'suggestedforyou'`,
      );
    });

    it('should throw error for unexpected Api status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      // @ts-expect-error intentionally break TS
      fetchMock.mockResponseOnce(null);

      await expect(
        fetchSuggestedForYouDetailsApiCall(
          'e9a06a8f-9d40-41b9-a8b4-38bbc67159a2',
        ),
      ).rejects.toThrow(`Perform a request failed`);
    });

    it('should throw error for other non-200 status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      fetchMock.mockResponseOnce(JSON.stringify({}), { status: 500 });

      await expect(
        fetchSuggestedForYouDetailsApiCall(
          'e9a06a8f-9d40-41b9-a8b4-38bbc67159a2',
        ),
      ).rejects.toThrow(
        `fetchSuggestedForYouDetailsApiCall(): failed to fetch data from endpoint 'suggestedforyou'`,
      );
    });
  });

  describe('useSuggestedForYouDetails', () => {
    it('should return suggestedForYou data and no error on successful fetch', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.suggestedforyou[0] }),
        {
          status: 200,
        },
      );

      mockRQ.mockReturnValueOnce({
        data: await fetchSuggestedForYouDetailsApiCall(
          'e9a06a8f-9d40-41b9-a8b4-38bbc67159a2',
        ),
        error: null,
        isLoading: false,
      });

      const { result } = renderHook(() =>
        useSuggestedForYouDetails({
          suggestedForYouContentId: 'e9a06a8f-9d40-41b9-a8b4-38bbc67159a2',
        }),
      );

      expect(result.current.data).toMatchSnapshot();
      expect(result.current.error).toBe(null);
      expect(result.current.isLoading).toBe(false);
    });

    it('should return isLoading as true before data loads', () => {
      mockRQ.mockReturnValueOnce({
        data: null,
        error: null,
        isLoading: true,
      });

      const { result } = renderHook(() =>
        useSuggestedForYouDetails({
          suggestedForYouContentId: 'e9a06a8f-9d40-41b9-a8b4-38bbc67159a2',
        }),
      );

      expect(result.current.isLoading).toBe(true);
      expect(result.current.data).toBe(null);
      expect(result.current.error).toBe(null);
    });

    test.each([
      'e9a06a8f-9d40-41b9-a8b4-38bbc67159a2',
      '23232',
      'some string',
      undefined,
      null,
    ])('should return error (case: %s)', (inputValue) => {
      const mockError = new Error('Network error');

      mockRQ.mockReturnValueOnce({
        data: null,
        error: mockError,
        isLoading: false,
      });

      const { result } = renderHook(() =>
        // @ts-expect-error intentionally break TS
        useSuggestedForYouDetails({ suggestedForYouContentId: inputValue }),
      );

      expect(result.current.error).toBe(mockError);
      expect(result.current.data).toBe(null);
      expect(result.current.isLoading).toBe(false);
    });
  });
});
