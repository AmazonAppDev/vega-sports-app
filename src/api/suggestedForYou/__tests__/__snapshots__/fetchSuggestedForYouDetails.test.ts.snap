// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchSuggestedForYouDetailsApiCall fetchSuggestedForYouDetailsApiCall should return parsed airDate 1`] = `
{
  "airDate": undefined,
  "description": "Marathon Star is a documentary about the life of a man who travels the world on a motorcycle. It is a story of adventure, freedom, and the pursuit of dreams. It is a story of a man who travels the world on a motorcycle.",
  "genre": "Documentary",
  "itemId": "empty-air-date",
  "network": "Bluejam",
  "rating": 2.7,
  "sportType": "volleyball",
  "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/900x700/6.jpg",
  "title": "Marathon Star",
  "videoSource": {
    "format": "HLS",
    "title": "Dash",
    "type": "hls",
    "uri": "https://storage.googleapis.com/shaka-demo-assets/bbb-dark-truths-hls/hls.m3u8",
  },
}
`;

exports[`fetchSuggestedForYouDetailsApiCall fetchSuggestedForYouDetailsApiCall should return parsed suggested for you data on success 1`] = `
{
  "airDate": 2020-11-14T04:52:43.000Z,
  "description": "American Vagabond is a documentary about the life of a man who travels the world on a motorcycle. It is a story of adventure, freedom, and the pursuit of dreams. It is a story of a man who travels the world on a motorcycle.",
  "genre": "Documentary",
  "itemId": "e9a06a8f-9d40-41b9-a8b4-38bbc67159a2",
  "network": "Bluejam",
  "rating": 2.7,
  "sportType": "volleyball",
  "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/900x700/1.jpg",
  "title": "American Vagabond",
  "videoSource": {
    "format": "HLS",
    "title": "NASA M3U8 stream",
    "type": "hls",
    "uri": "https://demo.unified-streaming.com/k8s/features/stable/video/tears-of-steel/tears-of-steel.ism/.m3u8",
  },
}
`;

exports[`fetchSuggestedForYouDetailsApiCall useSuggestedForYouDetails should return suggestedForYou data and no error on successful fetch 1`] = `
{
  "airDate": 2020-11-14T04:52:43.000Z,
  "description": "American Vagabond is a documentary about the life of a man who travels the world on a motorcycle. It is a story of adventure, freedom, and the pursuit of dreams. It is a story of a man who travels the world on a motorcycle.",
  "genre": "Documentary",
  "itemId": "e9a06a8f-9d40-41b9-a8b4-38bbc67159a2",
  "network": "Bluejam",
  "rating": 2.7,
  "sportType": "volleyball",
  "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/900x700/1.jpg",
  "title": "American Vagabond",
  "videoSource": {
    "format": "HLS",
    "title": "NASA M3U8 stream",
    "type": "hls",
    "uri": "https://demo.unified-streaming.com/k8s/features/stable/video/tears-of-steel/tears-of-steel.ism/.m3u8",
  },
}
`;
